src/Pull.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <functional>
+#include <memory>
+#include <nanomsg/pipeline.h>
+#include <nanomsg/nn.h>
+#include <string>
+#include <vector>
+#include "NanoProtocol.h"
+class Pull {
+public:
+   Pull(const std::string& location, const bool shouldConnect=false);
+   Pull(const std::string& location,
```

Comment:
@KjellKod  i will kjellify the constructors so they're simple with factory functions taking care of the work.